Configure sub-loggers through JSON appsettings

skomis-mm opened the pull request:
- fixes [#37](https://github.com/serilog/serilog-settings-configuration/issues/37)
- adds `Serilog.Sinks.Async` support
- adds `LoggingLevelSwitch` parameter type support

Basically implementation based on converters around `Action<TSerilogConfiguration>` parameter types from configuration extensions methods.

nblumhardt commented on src/Serilog.Settings.Configuration/Settings/Configuration/ConfigurationSectionArgumentValue.cs:
```diff
@@ -0,0 +1,39 @@
+                return new Action<LoggerConfiguration>(_configReader.Configure);
+            }
+
+            throw new Exception($"Handling {configurationType} is not implemented.");
```
Since this is wholly dependent on the value of `toType`, would `ArgumentException` be a reasonable choice here?

nblumhardt commented:
> adds LoggingLevelSwitch parameter type support

How does this one work in practice - is it possible to share a single level switch between the logger and one of the sinks (a-la Seq Sink's `controlLevelSwitch`)?

skomis-mm commented:
@nblumhardt , regarding `LoggingLevelSwitch`, I was thinking about defining switches and sharing them in the configuration. Not sure about it design so I did not included implementation for this.  

In this `pr` I just added converter from `string` to `LoggingLevelSwitch` for the extension method parameters types. For now it is just for `.Logger(..)` surrogate method that I added:
`internal static LoggerConfiguration Logger(LoggerSinkConfiguration loggerSinkConfiguration, Action<LoggerConfiguration> configureLogger, LoggingLevelSwitch restrictedToMinimumLevel = null)` so you can change level at runtime.

skomis-mm commented:
I've tested this for a while so I believe its ready to be in the wild. :bowtie:

nblumhardt commented:
Ah, I see now where `LoggingLevelSwitch` applies - thanks 👍

MelnikovIG commented:
@nblumhardt Add example of using this to documentation please.
https://github.com/serilog/serilog-settings-configuration

AraHaan commented:
Can ``Matching.FromSource`` be used in said configuration file when they want to filter with ``ByIncludingOnly``?
